File: Pb-4/pb-4.go
package main
import(
  "fmt"
  "math"
  "strconv"
)

func getPalindrome(n float64) int64{
  MAX := math.Pow(10, n) - 1
  big, bdiff, prod, diff := 0.0, 0.0, 0.0, 0.0
  start, skip := 0.0, 0.0
  if int(n) % 2 == 0{
    return int64((math.Pow(10, n / 2) - 1) * math.Pow(10, (3 * n) / 2) + math.Pow(10, n / 2) - 1)
  }else{
    s := ""
    for i := MAX; i >= 1; i--{
      start = MAX - 9
      skip = 11
      if int64(i) % 11 == 0{
        start = MAX
        skip = 1
      }
      for j:= start; j >= 1; j -= skip{
        prod = i * j
        diff = math.Abs(i - j)
        if prod > big && diff > bdiff{
          s = strconv.Itoa(int(prod))
          if s[0] == '9' && s[len(s) - 1] == '9' && isPal(int64(prod)){
            big = prod
          }
        }
      }
    }
  }
  return int64(big)
}

func isPal(n int64) bool{
  var rev , og int64 = 0, n
  for n > 0 {
    remainder := n % 10
    rev *= 10
    rev += remainder
    n /= 10
  }
  return og == rev
}

func main(){
  fmt.Println(getPalindrome(5))
}

File: Pb-4/pb-4.java
public class HelloWorld{

  public static void main(String []args){
        
    System.out.println(largestPalindromeProduct(2));  
  }

  public static double largestPalindromeProduct(int n){

    double max=0;
    
    double b= (Math.pow(10,n)-1);
    for(int i=(int)b;i>=1;i--){
        
      for(int j=i;j>=1;j--){
        n =i*j;
	if(n>max){

          String c=String.valueOf(n);
          StringBuilder c1 = new StringBuilder(); 
          c1.append(c); 
          c1 = c1.reverse();
          if(c.equals(c1.toString())){

	    max=n;

          }
	}      
      }
    }     
    return max; 
  }
}